clients/src/main/java/org/apache/kafka/clients/consumer/internals/AsyncKafkaConsumer.java

```diff
@@ -641,17 +641,15 @@ private void updateGroupMetadata(final Optional<Integer> memberEpoch, final Stri
 
     @Override
     public void registerMetricForSubscription(KafkaMetric metric) {
-        if (clientTelemetryReporter.isPresent()) {
-            ClientTelemetryReporter reporter = clientTelemetryReporter.get();
-            reporter.metricChange(metric);
+        if (!metrics().containsKey(metric.metricName())) {
+            clientTelemetryReporter.ifPresent(reporter -> reporter.metricChange(metric));
```

Comment:
I took another pass over the telemetry code - it generally shouldn't be a problem to re-register an existing metric but we shouldn't allow a "back door" to updating existing consumer metrics.

Reply:
> but we shouldn't allow a "back door" to updating existing consumer metrics.

Yes, that is my concern